v4.3.0 Changelog Update

StephenDaDev opened the pull request:
PR to update the changelog for the bot. Also updating project readme to include python 3.10-3.12.

StephenDaDev committed:
Update changelog and readme

StephenDaDev committed:
Add 3416, 3417, and 3418

StephenDaDev committed:
Add 3406, 3421, 3422, 3423 & Bump to 4.3.0 & Update other version refs

sebkuip commented on CHANGELOG.md:
```diff
@@ -6,6 +6,34 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
+- Ability to add attachments directly to snippets via storing image in MongoDB. [#3421](https://github.com/modmail-dev/Modmail/pull/3421)
+
+### Changed
+- Updated details on v4.2.1 changelog to better explain changes [#3409](https://github.com/modmail-dev/Modmail/pull/3409)
```
This line is pointing to an irrelevant PR that is closed. I think it might be the wrong link?

sebkuip commented on CHANGELOG.md:
```diff
@@ -6,6 +6,34 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
+
+### Changed
+- Updated details on v4.2.1 changelog to better explain changes [#3409](https://github.com/modmail-dev/Modmail/pull/3409)
+- Updated plugin registry [#3409](https://github.com/modmail-dev/Modmail/pull/3409)
```
This PR is closed. Did this get changed in another PR potentially?

sebkuip reviewed (changes requested):
2 lines point to #3409 which is closed. I think this might be a small oversight/issue

sebkuip committed:
Merge branch 'development' into users/stephen/4-2-2-changelog

StephenDaDev commented:
Yes, the issue was closed and replaced I have to figure out which one it is. Will push today hopefully

StephenDaDev committed:
Remove references to 3409

It appears that one of the changes was never implemented in a new PR, and one was in 3412 instead.

StephenDaDev committed:
Add #3425 (misc fixes for threadmenu)

StephenDaDev committed:
Update changelog to remove 3418 due to delay, and update readme with new version number

StephenDaDev committed:
Merge branch 'development' into users/stephen/4-2-2-changelog

sebkuip reviewed (dismissed):
All good

sebkuip committed:
Merge branch 'development' into users/stephen/4-2-2-changelog

StephenDaDev committed:
add 3430 and 3418 to changelog

StephenDaDev committed:
Merge branch 'development' into users/stephen/4-2-2-changelog

martinbndr commented on CHANGELOG.md:
```diff
@@ -6,6 +6,35 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
+- Bot events when threads is snoozed/unsnoozed [#3412](https://github.com/modmail-dev/Modmail/pull/3412)
+- Check for threadmenu and advancedmenu being enabled concurrently. [#3411](https://github.com/modmail-dev/Modmail/pull/3411)
+- Check for git installation before attempting autoupdate or update command. [#3406](https://github.com/modmail-dev/Modmail/pull/3406)
+- Check to see if bot is using Git repository for install before running update or autoupdate. [#3406](https://github.com/modmail-dev/Modmail/pull/3406)
```
```suggestion
- Check for git installation before attempting autoupdate or update command. [#3406](https://github.com/modmail-dev/Modmail/pull/3406)
```
Probably just added twice by accident. Just one needed ig.

StephenDaDev committed:
fix comments + devchat discussion

StephenDaDev committed:
add 3447

StephenDaDev committed:
Merge branch 'development' into users/stephen/4-2-2-changelog

Signed-off-by: Stephen <48072084+StephenDaDev@users.noreply.github.com>

sebkuip committed:
Add 2 fixes to the changelog

Updated CHANGELOG with recent bug fixes and improvements.

Signed-off-by: Sebastian <61157793+sebkuip@users.noreply.github.com>

sebkuip reviewed (dismissed):
All good now

lorenzo132 committed:
Merge branch 'development' into users/stephen/4-2-2-changelog

StephenDaDev committed:
add 3443

martinbndr reviewed (commented):
Looks good.

lorenzo132 committed:
Merge branch 'development' into users/stephen/4-2-2-changelog

sebkuip reviewed (approved):
No other comments since last review
